nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/EncryptAttributes.java

```diff
@@ -0,0 +1,508 @@
+            attrsToEncrypt.remove(CoreAttributes.FILENAME.key());
+            getLogger().info("Removing filename from {}cryption because of {} algorithm",
+                    new Object[]{(encrypt)?"en":"de", algorithm});
+        }
+        return attrsToEncrypt;
+    }
+
+    private Map<String, String> buildNewAttributes(FlowFile flowFile, PropertyValue attrList,
+                                                   PropertyValue attrSelectRegex, String algorithm,
+                                                   Encryptor encryptor, boolean encrypt) throws Exception {
+
+        Map<String, String> oldAttrs = flowFile.getAttributes();
+        Map<String, String> newAttrs = new HashMap<>();
+        Set<String> attrToEncrypt = getAttrToEncrypt(flowFile, attrList, attrSelectRegex, algorithm, encrypt);
+
+        for (String attr : attrToEncrypt) {
+            String attrVal = oldAttrs.get(attr);
+            String encryptedVal = (encrypt) ? performEncryption(attrVal, encryptor) : performDecryption(attrVal, encryptor);
+            newAttrs.put(attr, encryptedVal);
+            getLogger().debug("{}crypted {} from '{}' to '{}'",
+                    new Object[]{(encrypt)?"en":"de", attr, attrVal, encryptedVal});
+        }
+
+        return newAttrs;
+    }
+
+    @OnScheduled
+    public void onScheduled(final ProcessContext context) {
+        propMap = new HashMap<>();
+        for (PropertyDescriptor propDescriptor : context.getProperties().keySet()) {
+            if (propDescriptor.isDynamic()) {
+                propMap.put(propDescriptor.getName(), context.getProperty(propDescriptor));
+                getLogger().info("Adding dynamic property: {}", new Object[]{propDescriptor});
+            }
+        }
+    }
+
+    @Override
+    public void onTrigger(ProcessContext context, ProcessSession session) throws ProcessException {
+
+        FlowFile flowFile = session.get();
+        if (flowFile == null) {
+            return;
+        }
+
+        final ComponentLog logger = getLogger();
+        final PropertyValue attrList = context.getProperty(ATTRS_TO_ENCRYPT);
+        final PropertyValue attrSelectRegex = context.getProperty(ATTR_SELECT_REG_EX);
+        final String method = context.getProperty(ENCRYPTION_ALGORITHM).getValue();
+        final EncryptionMethod encryptionMethod = EncryptionMethod.valueOf(method);
+        final String providerName = encryptionMethod.getProvider();
+        final String algorithm = encryptionMethod.getAlgorithm();
+        final String password = context.getProperty(PASSWORD).getValue();
+        final KeyDerivationFunction kdf = KeyDerivationFunction.valueOf(context.getProperty(KEY_DERIVATION_FUNCTION).getValue());
+        final boolean encrypt = context.getProperty(MODE).getValue().equalsIgnoreCase(ENCRYPT_MODE);
+
+        Encryptor encryptor;
+        Map<String, String> newAtrList;
+
+        try {
+            if (EncryptProcessorUtils.isPGPAlgorithm(algorithm)) {
```

Comment:
This `try` block is duplicated from `EncryptContent` -- can be refactored to a shared method.